frontend/services/entities/mdm.ts

```diff
@@ -72,24 +114,62 @@ export default {
     return sendRequest("DELETE", MDM_PROFILE(profileId));
   },
 
-  getAggregateProfileStatuses: (teamId = APP_CONTEXT_NO_TEAM_ID) => {
+  // TODO: API INTEGRATION: we need to rework this when we create API call that
+  // will return the aggregate statuses for windows included in the response.
+  // Currently to get windows data included we will need to make a separate call.
+  // We will likely change this to go back to single "getProfileStatusSummary" API call.
+  getAggregateProfileStatuses: async (
+    teamId = APP_CONTEXT_NO_TEAM_ID,
+    // TODO: WINDOWS FEATURE FLAG: remove when we windows feature is released.
+    includeWindows: boolean
+  ) => {
+    // if we are not including windows we can just call the existing profile summary API
+    if (!includeWindows) {
+      return mdmService.getProfileStatusSummary(teamId);
+    }
+
+    // otherwise we have to make two calls and combine the results.
+    const [
+      profileStatuses,
+      diskEncryptionSummary,
+    ] = await mdmService.getAggregateProfileStatusesWithWindows(teamId);
+
+    return new Promise<ProfileStatusSummaryResponse>((resolve) => {
+      resolve(
+        generateCombinedProfileStatusSummary(
+          profileStatuses,
+          diskEncryptionSummary
+        )
+      );
+    });
+  },
+
+  getAggregateProfileStatusesWithWindows: async (teamId: number) => {
+    const [profileStatuses, diskEncryptionSummary] = await Promise.all([
+      mdmService.getProfileStatusSummary(teamId),
+      mdmService.getDiskEncryptionSummary(teamId),
+    ]);
+    return [profileStatuses, diskEncryptionSummary];
+  },
+
```

Comment:
Not sure if this makes is more or less clear, but what do you think of this approach?

```ts
  getAggregateProfileStatuses: async (
    teamId = APP_CONTEXT_NO_TEAM_ID,
    includeWindows: boolean
  ): Promise<ProfileStatusSummaryResponse> => {
    // if we are not including windows we can just call the existing profile summary API
    if (!includeWindows) {
      return mdmService.getProfileStatusSummary(teamId);
    }

    return mdmService
      .getAggregateProfileStatusesWithWindows(teamId)
      .then((res) => generateCombinedProfileStatusSummary(...res));
  },

  getAggregateProfileStatusesWithWindows: (teamId: number) => {
    return Promise.all([
      mdmService.getProfileStatusSummary(teamId),
      mdmService.getDiskEncryptionSummary(teamId),
    ]);
  },

```

To make it work for TS, I think you may need to also specify the return types of `getProfileStatusSummary` as `Promise<ProfileStatusSummaryResponse> ` and `getDiskEncryptionSummary` as `Promise<IDiskEncryptionSummaryResponse> `.

Reply:
hey yes this easier to reason about. gonna change it to this.